encoders/numeric/CompressionVaeEncoder/__init__.py

```diff
@@ -67,14 +67,14 @@ def post_init(self):
 
                 self.to_device()
 
-    @batching()
+    @batching
     @as_ndarray
     def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
         """
         :param data: a `B x T` numpy ``ndarray``, `B` is the size of the batch
         :return: a `B x D` numpy ``ndarray``
         """
         import tensorflow as tf
-
-        return self.sess.run([self.embeddings],
-                             feed_dict={self.data_feature_placeholder: data})[0]
+        with self.sess as sess:
```

Comment:
Would that contrxg manager revuild the session? Would it be more effective to have only one during executor lifetime?

Reply:
Should I use @cached_property to cache it?

Reply:
No, I do not think so, my point is one only needs to close it once